src/bridge/cockpitpipechannel.c

```diff
@@ -384,6 +407,10 @@ cockpit_pipe_channel_prepare (CockpitChannel *channel)
       else
         self->pipe = cockpit_pipe_spawn ((const gchar **)argv, (const gchar **)env, dir, flags);
     }
+  else if (internal && lookup_internal_fd (internal, &fd))
+    {
+      self->pipe = cockpit_pipe_new (internal, fd, fd);
```

Comment:
The only way known to me is to check if `fdopen()` with a  particular mode returns `EINVAL` or succeeds. But I guess I may have misunderstood this -- so you are saying that cockpit itself should not touch/use the fd at all, just pass it between the service and the client? That would actually make a lot of sense, I just got misled when reviewing this.

Do I understand this alright -- with this, when the client opens a `cockpit.channel()` on this handle, it's still the client's responsibility to know whether it can read and/or write into this? If so, then we should really have a test case that ensures that cockpit doesn't touch the fds themselves -- i. e. trying to write to a read fd throws a sensible error, and we should have a test case for the write end of a pipe only to ensure that this direction works as well.